### ISSUE TITLE
Failed to download vs-code extension

### ISSUE THREAD

**Issue by parroty** [advanced-security/grab_ql#16]
Hi 👋 . I was experimenting with this `grab-codeql` command and faced an error, and just wanted to ask for your opinion.

## Behavior

`grab-codeql` command fails with `ERROR:grab_codeql.grab_codeql:Didn't get extension version informatio` error. The following is a result from the linux environment (`Ubuntu 20.04.4 LTS`).

```
$ wget https://github.com/advanced-security/grab_ql/releases/download/0.0.3/grab_codeql-0.0.3-py3-none-any.whl
$ python3 -m pip install grab_codeql-*-py3-none-any.whl
$ grab-codeql
INFO:grab_codeql.grab_codeql:CLI tag is v2.10.3, getting for platform linux/intel/64
codeql-linux64.zip: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████| 413M/413M [00:04<00:00, 103MB/s]
INFO:grab_codeql.grab_codeql:Library tag is codeql-cli/v2.10.3
github-codeql-codeql-cli-latest-0-gf30b735.zip: 100%|██████████████████████████████████████████████████████████████████████| 38.7M/38.7M [00:01<00:00, 22.1MB/s]
code-stable-x64-1660629670.tar.gz: 100%|█████████████████████████████████████████████████████████████████████████████████████| 114M/114M [00:01<00:00, 84.7MB/s]
ERROR:grab_codeql.grab_codeql:JSON error: Expecting value: line 3 column 1 (char 4)
ERROR:grab_codeql.grab_codeql:Didn't get extension version information
ERROR:grab_codeql.grab_codeql:Failed to get latest item: max() arg is an empty sequence
ERROR:grab_codeql.grab_codeql:🔥 VSCode extension download failed. Please check the arguments you passed in, try https://marketplace.visualstudio.com/items?itemName=GitHub.vscode-codeql or report the error in an issue.
```

It seems the error is coming from the VS Code extension, since the following command succeeds.

```
$ grab-codeql --no-vscode-ext
INFO:grab_codeql.grab_codeql:CLI tag is v2.10.3, getting for platform linux/intel/64
codeql-linux64.zip: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████| 413M/413M [00:14<00:00, 29.9MB/s]
INFO:grab_codeql.grab_codeql:Library tag is codeql-cli/v2.10.3
github-codeql-codeql-cli-latest-0-gf30b735.zip: 100%|██████████████████████████████████████████████████████████████████████| 38.7M/38.7M [00:01<00:00, 27.5MB/s]
code-stable-x64-1660629670.tar.gz: 100%|██
```

## Possible Cause

It seems the following API endpoint is called, 
- `https://marketplace.visualstudio.com/public%2Fgallery%2Fextensionquery?api-version%3D3.0-preview.1`

but I assume the expected endpoint would be the following.
- `https://marketplace.visualstudio.com/_apis/public/gallery/extensionquery?api-version=3.0-preview.1`

It would be related to the following logic.

- https://github.com/advanced-security/grab_ql/blob/main/grab_codeql/grab_codeql.py#L329

```
$ python3
Python 3.10.4 (main, Aug 11 2022, 09:50:42) [GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from urllib.parse import urljoin, quote_plus
>>> urljoin("https://marketplace.visualstudio.com/_apis", f"{quote_plus('public/gallery/extensionquery')}?{quote_plus('api-version=3.0-preview.1')}")
'https://marketplace.visualstudio.com/public%2Fgallery%2Fextensionquery?api-version%3D3.0-preview.1'
```

## Possible Solution (not confident enough, though)

I experimented with the following change and at least succeeded in my environment.

```diff
diff --git a/grab_codeql/grab_codeql.py b/grab_codeql/grab_codeql.py
index 43a63f5..67db735 100755
--- a/grab_codeql/grab_codeql.py
+++ b/grab_codeql/grab_codeql.py
@@ -130,8 +130,8 @@ MARKETPLACE_FILTERTYPE_TARGET = 8
 MARKETPLACE_ASSETTYPE_VSIX = "Microsoft.VisualStudio.Services.VSIXPackage"
 MARKETPLACE_VSCODE_TARGET = "Microsoft.VisualStudio.Code"
 MARKETPLACE_CODEQL_FQNAME = "GitHub.vscode-codeql"
-MARKETPLACE_API_BASE = "https://marketplace.visualstudio.com/_apis"
-MARKETPLACE_API_EXTENSION_ENDPOINT = "/public/gallery/extensionquery"
+MARKETPLACE_API_BASE = "https://marketplace.visualstudio.com/_apis/"
+MARKETPLACE_API_EXTENSION_ENDPOINT = "public/gallery/extensionquery"
 MARKETPLACE_API_VERSION = "api-version=3.0-preview.1"
 MARKETPLACE_HTTP_ACCEPT = f"application/json; {MARKETPLACE_API_VERSION}"
 
@@ -327,7 +327,7 @@ class MarketPlaceApi():
         """Init API."""
         self._uri = urljoin(
             MARKETPLACE_API_BASE,
-            f"{quote_plus(MARKETPLACE_API_EXTENSION_ENDPOINT)}?{quote_plus(MARKETPLACE_API_VERSION)}"
+            f"{MARKETPLACE_API_EXTENSION_ENDPOINT}?{MARKETPLACE_API_VERSION}"
         )
         self._headers = CaseInsensitiveDict({
             HTTP_HEADER_CONTENT_TYPE: HTTP_APPLICATION_JSON,
```

**Comment by aegilops** [comment]
Thanks! Can you try out release 0.0.4 and see if it's fixed for you please?

Your analysis was correct - I'd got carried away with quoting things I didn't need to.

**Comment by aegilops** [comment]
Yep, I've reproduced that. Thanks, I'll see what's wrong and how to improve my tests to catch that - I thought I'd dealt with that problem.

**Comment by aegilops** [comment]
The fixes I made to correct this aren't there: I mustn't have merged the fixing branch correctly, so I'll have to redo the fix now.

**Comment by aegilops** [comment]
Fixed in the amended version of the 0.0.4 release.

I've tested it with the now-functioning `--dry-run` and a full download on MacOS and Linux.

I re-did the release for `v0.0.4`, so please grab that and use `--force-reinstall` when you reinstall.

Thanks for your report and your patience!

Test:
```
% grab-codeql --vscode-macos-installer zip --dry-run
INFO:grab_codeql.grab_codeql:✅ CLI tag is v2.10.4, getting for platform macos/arm/64
INFO:grab_codeql.grab_codeql:ℹ️  Ending download, dry-run only. Would have got 411368927B to codeql-osx64.zip
INFO:grab_codeql.grab_codeql:✅ Library tag is codeql-cli/v2.10.4
INFO:grab_codeql.grab_codeql:ℹ️  Ending download, dry-run only. Would have got 0B to github-codeql-codeql-cli-latest-0-gf318dd5.zip
INFO:grab_codeql.grab_codeql:✅ VSCode version is latest, getting for platform macos/arm/64
INFO:grab_codeql.grab_codeql:ℹ️  Ending download, dry-run only. Would have got 107956080B to VSCode-darwin-arm64.zip
INFO:grab_codeql.grab_codeql:✅ VSCode extension version is 1.6.12, getting
INFO:grab_codeql.grab_codeql:ℹ️  Ending download, dry-run only. Would have got 16557519B to GitHub.vscode-codeql_1.6.12.vsix
```

Full run:
```
% grab-codeql --vscode-macos-installer zip          
INFO:grab_codeql.grab_codeql:✅ CLI tag is v2.10.4, getting for platform macos/arm/64
codeql-osx64.zip: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 392M/392M [00:36<00:00, 11.3MB/s]
INFO:grab_codeql.grab_codeql:✅ Library tag is codeql-cli/v2.10.4
github-codeql-codeql-cli-latest-0-gf318dd5.zip: 39.8MB [00:06, 6.58MB/s]
INFO:grab_codeql.grab_codeql:✅ VSCode version is latest, getting for platform macos/arm/64
VSCode-darwin-arm64.zip: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 103M/103M [00:09<00:00, 11.7MB/s]
INFO:grab_codeql.grab_codeql:✅ VSCode extension version is 1.6.12, getting
GitHub.vscode-codeql_1.6.12.vsix: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 15.8M/15.8M [00:01<00:00, 11.4MB/s]
```

**Comment by parroty** [comment]
Thanks! It worked in my environment nicely as well 🙇
